drivers/sensor/xbr818/xbr818.c

```diff
@@ -0,0 +1,450 @@
+		return -EINVAL;
+	}
+
+	ret = xbr818_enable_i2c(dev);
+	if (ret != 0) {
+		return ret;
+	}
+
+	/* this cast is necessary due to the additional attributes not being part of the
+	 * initial enum and causing a build error without it.
+	 */
+	switch ((int)attr) {
+	case SENSOR_ATTR_LOWER_THRESH:
+		if (val->val1 > 0xFFFF) {
+			return -EINVAL;
+		}
+		tmp[0] = val->val1 & 0xFF;
+		tmp[1] = (val->val1 & 0xFF00) >> 8;
+		ret = i2c_burst_write_dt(&config->i2c, XBR818_THRESHOLD_1, tmp, 2);
+		break;
+	case SENSOR_ATTR_XBR818_NOISE_FLOOR:
+		if (val->val1 > 0xFFFF) {
+			return -EINVAL;
+		}
+		tmp[0] = val->val1 & 0xFF;
+		tmp[1] = (val->val1 & 0xFF00) >> 8;
+		ret = i2c_burst_write_dt(&config->i2c, XBR818_THRESHOLD_NOISE_1, tmp, 2);
+		break;
+	case SENSOR_ATTR_XBR818_DELAY_TIME:
+		tmpf = (uint64_t)val->val1 * 1000000 + (uint64_t)val->val2;
+		tmpf = (tmpf * SENSOR_XBR818_CLOCKRATE) / 1000000;
+		if (tmpf > 0xFFFFFF) {
+			return -EINVAL;
+		}
+		tmp[0] = tmpf & 0xFF;
+		tmp[1] = (tmpf & 0xFF00) >> 8;
+		tmp[2] = (tmpf & 0xFF0000) >> 16;
+		ret = i2c_burst_write_dt(&config->i2c, XBR818_DELAY_TIME_1, tmp, 3);
+		break;
+	case SENSOR_ATTR_XBR818_LOCK_TIME:
+		tmpf = (uint64_t)val->val1 * 1000000 + (uint64_t)val->val2;
+		tmpf = (tmpf * SENSOR_XBR818_CLOCKRATE) / 1000000;
+		if (tmpf > 0xFFFFFF) {
+			return -EINVAL;
+		}
+		tmp[0] = tmpf & 0xFF;
+		tmp[1] = (tmpf & 0xFF00) >> 8;
+		tmp[2] = (tmpf & 0xFF0000) >> 16;
+		ret = i2c_burst_write_dt(&config->i2c, XBR818_LOCK_TIME_1, tmp, 3);
+		break;
+	case SENSOR_ATTR_XBR818_RF_POWER:
+		if (val->val1 > 0x7) {
+			return -EINVAL;
+		}
+		tmp[0] = val->val1 & 0x7;
+		ret = i2c_reg_write_byte_dt(&config->i2c, XBR818_RF_POWER, tmp[0]);
+		break;
+	case SENSOR_ATTR_SAMPLING_FREQUENCY:
+		if (val->val1 > SENSOR_XBR818_CLOCKRATE) {
+			return -EINVAL;
+		}
+		tmp[0] = SENSOR_XBR818_CLOCKRATE / val->val1;
+		if (tmp[0] > 0xFF) {
+			return -EINVAL;
+		}
+		ret = i2c_reg_write_byte_dt(&config->i2c, XBR818_SAMPLE_RATE_DIVIDER, tmp[0]);
+		break;
+	default:
+		ret = -ENODEV;
+
+		break;
+	}
+
+	if (ret != 0) {
+		return ret;
+	}
+
+	ret = xbr818_disable_i2c(dev);
+	if (ret != 0) {
+		return ret;
+	}
+
+	return ret;
+}
+
+static int xbr818_attr_get(const struct device *dev, enum sensor_channel chan,
+			   enum sensor_attribute attr, struct sensor_value *val)
+{
+	const struct xbr818_config *config = dev->config;
+	int ret;
+	uint8_t tmp[3];
+	uint64_t tmpf;
+
+	if (chan != SENSOR_CHAN_PROX) {
+		LOG_ERR("%s: requesting unsupported channel %i", dev->name, chan);
+		return -ENOTSUP;
+	}
+
+	ret = xbr818_enable_i2c(dev);
+	if (ret != 0) {
+		return ret;
+	}
+
+	/* this cast is necessary due to the additional attributes not being part of the
+	 * initial enum and causing a build error without it.
+	 */
+	switch ((int)attr) {
+	case SENSOR_ATTR_LOWER_THRESH:
+		ret = i2c_burst_read_dt(&config->i2c, XBR818_THRESHOLD_1, tmp, 2);
+		if (ret != 0) {
+			break;
+		}
+		val->val1 = tmp[0] & 0xFF;
+		val->val1 |= tmp[1] << 8;
```

Comment:
Can you take a second look here. I think that all these `|=` assignment don't do anything. Since `tmp` is `uint8_t` isn't `tmp[1] << 8` just 0? I think you need to cast `(uint32_t)tmp[1]` before doing the shift.

Reply:
I have tested this functionality before pushing. It works because of [integer promotions](https://stackoverflow.com/questions/46073295/implicit-type-promotion-rules). Do you want it explicited?

Reply:
I think it needs to be explicit, yes (MISRA rule 10.6)

Reply:
Done.